crates/walrus-service/src/client/cli/internal_run.rs

```diff
@@ -447,6 +474,84 @@ fn apply_common_store_child_args(
     }
 }
 
+/// Spawns a task to forward signals to the child uploader process.
+/// This is used to forward the signals from the parent process to the child process.
+#[cfg(unix)]
+fn spawn_signal_forwarders(handle: ChildProcessHandle) {
+    use tokio::signal::unix::{SignalKind, signal};
+
+    let signals = [
+        (SignalKind::terminate(), libc::SIGTERM, "SIGTERM"),
+        (SignalKind::interrupt(), libc::SIGINT, "SIGINT"),
+    ];
+
+    for (kind, signo, name) in signals {
+        let handle_clone = handle.clone();
+        match signal(kind) {
+            Ok(stream) => {
+                let mut stream = stream;
+                tokio::spawn(async move {
```

Comment:
question: say the child process finishes, but the main process is a long running process, this thread will always stay until the main process quits, is that correct? I wonder if there is a way to abort this thread if the child process finishes.